radis/lbl/base.py

```diff
@@ -660,17 +660,29 @@ def get_Evib_CDSD_pcN_1iso(df, iso):
             # the map below is crazy fast compared to above loop
             df["Evibu"] = df_pcN.index.map(Evib_dict.get).values
 
-            return df.loc[idx, ["Evibl", "Evibu"]]
+            try:
+                return df.loc[idx, ["Evibl", "Evibu"]]
+            except:
+                return df.loc[df.index.tolist(), ["Evibl", "Evibu"]]
 
         #        df = df.groupby('iso').apply(lambda x: get_Evib_CDSD_pcN_1iso(x, x.name))
 
         df["Evibl"] = np.nan
         df["Evibu"] = np.nan
-        for iso, idx in df.groupby("iso").indices.items():
-            df.loc[idx, ["Evibl", "Evibu"]] = get_Evib_CDSD_pcN_1iso(df.loc[idx], iso)
+        if "iso" in df:
+            for iso, idx in df.groupby("iso").indices.items():
+                df.loc[idx, ["Evibl", "Evibu"]] = get_Evib_CDSD_pcN_1iso(
+                    df.loc[idx], iso
+                )
 
-            if radis.DEBUG_MODE:
-                assert (df.loc[idx, "iso"] == iso).all()
+                if radis.DEBUG_MODE:
+                    assert (df.loc[idx, "iso"] == iso).all()
+
+        else:
+            iso = df.attrs["iso"]
+            df.loc[df.index.tolist(), ["Evibl", "Evibu"]] = get_Evib_CDSD_pcN_1iso(
+                df.loc[df.index.tolist()], iso
+            )
```

Comment:
same, here you can probably try : 
```suggestion
            df.loc[:, ["Evibl", "Evibu"]] = get_Evib_CDSD_pcN_1iso(df, iso)
```